### PR TITLE
Make kafka handler able to propagate

### PR DESCRIPTION
**Pull request by xinyaohu**
This makes KafkaHandler able to propagte the stream so that we could chain things together. 

@alexism

### CODE REVIEW

**Comment by xxiao23** [comment]
:+1:

**Comment by xinyaohu** [comment]
@alexism i will proceed to separate our plog, please help me review this PR

File: plog-kafka/src/main/java/com/airbnb/plog/kafka/KafkaHandler.java [hunk 1]
```diff
@@ -77,6 +86,7 @@ public JsonObject getStats() {
         return new JsonObject()
                 .add("default_topic", defaultTopic)
                 .add("seen_messages", seenMessages.get())
+                .add("propagated_messages", propagatedMessages.get())
```
Comment (alexism) [inline plog-kafka/src/main/java/com/airbnb/plog/kafka/KafkaHandler.java]: The `propagate` flag value is the same for all messages, to `seen_messages` and `propagated_messages` will always be the same. amirite?
I'd then remove than metric.

File: plog-kafka/src/main/java/com/airbnb/plog/kafka/KafkaHandler.java [hunk 2]
```diff
@@ -57,6 +60,12 @@ protected void channelRead0(ChannelHandlerContext ctx, Message msg) throws Excep
         if (!sawKtTag) {
             sendOrReportFailure(defaultTopic, payload);
         }
+
+        if (propagate) {
+            propagatedMessages.incrementAndGet();
+            msg.retain();
```
Comment (alexism) [inline plog-kafka/src/main/java/com/airbnb/plog/kafka/KafkaHandler.java]: where is `msg.release()` invoked?

**Comment by alexism** [comment]
LGTM

**Comment by xinyaohu** [comment]
i will remove the metrics then. i will work on chef change first to separate our plog deployment. after that, we can merge this and test on prod

**Comment by xinyaohu** [comment]
removed the metric

**Pull request merged**